[Synthetics] Update infrastructure monitoring for k8s (#6367)
elastic/docs-content: solutions/observability/synthetics/scale-architect-synthetics-deployment.md

The Synthetics UI does *not* work with CCS/CCR because it would limit the rich user experience that the Synthetics UI provides. Unlike the majority of {{kib}} apps, the Synthetics UI relies heavily on state stored in {{kib}} shared objects in order to provide a rich user experience. Because {{kib}} saved objects cannot be shared via CCS/CCR, the Synthetics UI will not show any user data even if CCS/CCR is configured.

> ## Do not use the Synthetics UI for infrastructure or {{k8s}} monitoring [synthetics-no-k8s-infra]

Reviewer: Just one abstract suggestion, @alaudazzi take it as you may.

While the sentiment is correct, the title might come off as a little strong. K8s and infra monitoring is possible, the only thing that isn't is autodiscovery. Maybe that's what this should focus on and mention k8s as a collateral.

There might be users out there who are capable of managing target discovery on their end and just automate the monitor creation process through API and that's a completely valid case

Reviewer 2: Good point @emilioalvap , based on https://github.com/elastic/docs-content/issues/1879 I was thinking we didn't want users to use synthetics UI for infra monitoring in general (node up / down), besides Kubernetes monitoring with autodiscovery.

But if this is mainly about not having autodiscovery functionality that's a different story.

@alaudazzi , I'll leave it up to you to decide the best way of moving this forward! I agree with softening the note a bit, and trying to not include a title with a negative statement (do not do xxx)

> The {{synthetics-app}} is designed for active synthetic checks against user-defined URLs and user journeys. **Do not use it for host or pod availability monitoring via autodiscovery** — for example, automatically checking every pod in a {{k8s}} cluster as it starts and stops.

Reviewer 2: ```suggestion
The {{synthetics-app}} is designed for active synthetic checks against user-defined URLs and user journeys. It is not intended for infrastructure or {{k8s}} pod monitoring through autodiscovery.
```

The Synthetics UI only shows monitors that are explicitly created and managed through the [Synthetics UI](/solutions/observability/synthetics/create-monitors-ui.md) or a [Synthetics project](/solutions/observability/synthetics/create-monitors-with-projects.md). It has no mechanism for dynamic autodiscovery of infrastructure targets, and it is not designed to ingest or display the high volume of short-lived monitor results that infrastructure monitoring typically produces.

For infrastructure or {{k8s}} uptime monitoring, use one of the following approaches instead:

> * **{{heartbeat}} with autodiscovery**: Run {{heartbeat}} directly on your infrastructure and use its [autodiscovery](https://www.elastic.co/guide/en/beats/heartbeat/current/configuration-autodiscover.html) capabilities to dynamically monitor hosts and pods. Results appear in the [{{uptime-app}}](/solutions/observability/uptime/index.md).

Reviewer 2: ```suggestion
* **[{{heartbeat}}](/reference/heartbeat/index.md) with autodiscovery**: Run {{heartbeat}} on your infrastructure and use [autodiscovery](/reference/heartbeat/configuration-autodiscover.md) to dynamically monitor hosts and pods. Results appear in the [{{uptime-app}}](/solutions/observability/uptime/index.md).
```

> * **{{agent}} with the Uptime Monitors integration**: Deploy a standalone {{agent}} and configure the Uptime Monitors ({{heartbeat}}) integration to collect availability data from your infrastructure. Note that the Uptime app is deprecated as of 8.15. 

Reviewer 2: ```suggestion
* **[{{agent}}](/reference/fleet/install-elastic-agents.md) with the Uptime Monitors ({{heartbeat}}) integration**: Deploy {{agent}} and configure the Uptime Monitors integration to collect infrastructure availability data. For [Kubernetes autodiscovery](/reference/fleet/elastic-agent-kubernetes-autodiscovery.md), hints-based autodiscovery requires standalone mode. Note that the Uptime app is deprecated as of 8.15.
```
